homeassistant/components/ecobee/__init__.py

```diff
@@ -81,13 +83,11 @@ class EcobeeData:
     Also handle refreshing tokens and updating config entry with refreshed tokens.
     """
 
-    def __init__(self, hass, entry, api_key, refresh_token):
+    def __init__(self, hass, entry, config: dict = None):
         """Initialize the Ecobee data object."""
         self._hass = hass
         self._entry = entry
-        self.ecobee = Ecobee(
-            config={ECOBEE_API_KEY: api_key, ECOBEE_REFRESH_TOKEN: refresh_token}
-        )
+        self.ecobee = Ecobee(config=config)
```

Comment:
We shouldn't couple our api and the ecobee library api directly by using the same constant keys. Please explicitly use separate keys for the ecobee config.